MobileCenter/MobileCenter/Internals/Storage/MSDBStorage.m

```diff
@@ -14,31 +14,48 @@ - (instancetype)init {
   if (self) {
     _connection = [[MSSqliteConnection alloc] initWithDatabaseFilename:kMSDBFileName];
     _batches = [NSMutableDictionary<NSString *, NSArray<NSString *> *> new];
-    [self initTables];
+    _capacity = NSIntegerMax;
+
+    // Create the DB.
+    NSString *createLogTableQuery =
+        [NSString stringWithFormat:@"CREATE TABLE IF NOT EXISTS %@ (%@ INTEGER PRIMARY KEY AUTOINCREMENT, %@ TEXT NOT "
+                                   @"NULL, %@ TEXT NOT NULL);",
+                                   kMSLogTableName, kMSIdColumnName, kMSGroupIdColumnName, kMSDataColumnName];
+    [self.connection executeQuery:createLogTableQuery];
   }
   return self;
 }
 
-- (void)initTables {
-  NSString *createLogTableQuery = [NSString
-      stringWithFormat:
-          @"CREATE TABLE IF NOT EXISTS %@ (%@ INTEGER PRIMARY KEY AUTOINCREMENT, %@ TEXT NOT NULL, %@ TEXT NOT NULL);",
-          kMSLogTableName, kMSIdColumnName, kMSGroupIdColumnName, kMSDataColumnName];
-  [self.connection executeQuery:createLogTableQuery];
+- (instancetype)initWithCapacity:(NSInteger)capacity {
+  self = [self init];
+  if (self) {
+    _capacity = capacity;
+  }
+  return self;
 }
 
 #pragma mark - Save logs
 
 - (BOOL)saveLog:(id<MSLog>)log withGroupId:(NSString *)groupId {
+  BOOL succeeded = NO;
```

Comment:
You can move this to line 51. The variable doesn't need to be declared until it is used.